sdk/src/client/api/block_builder/input_selection/requirement/context_inputs.rs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{Error, InputSelection};
+use crate::{
+    client::secret::types::InputSigningData,
+    types::block::{
+        context_input::{BlockIssuanceCreditContextInput, CommitmentContextInput, RewardContextInput},
+        output::{AccountId, DelegationOutputBuilder, Output},
+    },
+};
+
+impl InputSelection {
+    pub(crate) fn fulfill_context_inputs_requirement(&mut self) -> Result<Vec<InputSigningData>, Error> {
+        let mut needs_commitment_context = false;
+
+        for (idx, input) in self.selected_inputs.iter().enumerate() {
+            // Transitioning an issuer account requires a BlockIssuanceCreditContextInput.
+            if let Output::Account(account) = &input.output {
+                if account.features().block_issuer().is_some() {
+                    log::debug!("Adding block issuance context input for transitioned account output");
+                    self.context_inputs.insert(
+                        BlockIssuanceCreditContextInput::from(account.account_id_non_null(input.output_id())).into(),
+                    );
+                }
+            }
+            // Transitioning an implicit account requires a BlockIssuanceCreditContextInput.
+            if let Output::Basic(implicit_account) = &input.output {
+                if implicit_account.is_implicit_account() {
```

Comment:
probably would keep calling it `basic` as long as we don't know whether it's actually an implicit account, but up to you.

Reply:
makes sense to me

Reply:
🤷‍♀️